src/internal/document_api/requests.rs

```diff
@@ -251,16 +259,19 @@ pub mod document_create {
                         shared_with: req_grants,
                     },
                 };
-                Box::new(auth.request.post(
-                    "documents",
-                    &req,
-                    RequestErrorCode::DocumentCreate,
-                    AuthV2Builder::new(&auth, Utc::now()),
-                ))
+                auth.request
+                    .post(
+                        "documents",
+                        &req,
+                        RequestErrorCode::DocumentCreate,
+                        AuthV2Builder::new(&auth, Utc::now()),
+                    )
+                    .compat()
+                    .await
             }
             // the failure case here is that we couldn't convert the recrypt EncryptedValue because
             // it was not an EncryptedOnceValue -- really just a limitation of Rust's enums as we expect these to be EncryptedOnceValues
-            Err(e) => Box::new(futures::future::failed(e)),
+            Err(e) => futures3::future::err(e).await,
```

Comment:
You can get rid of this branch entirely by putting a `?` on the end of line 251.

Reply:
I couldn't get a `?` on the end of the `collect` to be happy with type inference, but I did simplify this with a very similar outcome.